pkg/deploy/controller/deployment/controller.go

```diff
@@ -136,8 +137,55 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		}
 		c.recorder.Eventf(deployment, "cancelled", "Cancelled deployment")
 	case deployapi.DeploymentStatusFailed:
```

Comment:
Still reading the actual code here, but just wanted to point out that the Handle function is getting massive... should we start breaking out functions where possible?